components/support/jwcrypto/src/ec.rs

```diff
@@ -37,42 +45,14 @@ pub(crate) fn encrypt_to_jwe(
         apu: None,
         apv: None,
     };
-
     let secret = derive_shared_secret(&protected_header, local_key_pair, &ec_key_params)?;
-
-    let encryption_algorithm = match protected_header.enc {
-        EncryptionAlgorithm::A256GCM => &aead::AES_256_GCM,
-    };
-    let sealing_key = aead::SealingKey::new(encryption_algorithm, &secret.as_ref())?;
-    let additional_data = serde_json::to_string(&protected_header)?;
-    let additional_data =
-        base64::encode_config(additional_data.as_bytes(), base64::URL_SAFE_NO_PAD);
-    let additional_data = additional_data.as_bytes();
-    let aad = aead::Aad::from(additional_data);
-    let mut iv: Vec<u8> = vec![0; 12];
-    rand::fill(&mut iv)?;
-    let nonce = aead::Nonce::try_assume_unique_for_key(encryption_algorithm, &iv)?;
-    let mut encrypted = aead::seal(&sealing_key, nonce, aad, data)?;
-
-    let tag_idx = encrypted.len() - encryption_algorithm.tag_len();
-    let auth_tag = encrypted.split_off(tag_idx);
-    let ciphertext = encrypted;
-
-    CompactJwe::new(
-        Some(protected_header),
-        None,
-        Some(iv),
-        ciphertext,
-        Some(auth_tag),
-    )
+    aes::aes_gcm_encrypt(data, protected_header, secret.as_ref())
 }
 
-pub(crate) fn decrypt_jwe(
-    jwe: &CompactJwe,
-    decryption_params: DecryptionParameters,
-) -> Result<String> {
-    let DecryptionParameters::ECDH_ES { local_key_pair } = decryption_params;
-
+/// The ECDH helper that takes the ciphertext in the form of a CompactJwe,
+/// and the keys, creates the appropriate header, then calls the `aes` module to
+/// do the actual decrytion.
+pub(crate) fn decrypt_jwe(jwe: &CompactJwe, local_key_pair: EphemeralKeyPair) -> Result<String> {
     let protected_header = jwe.protected_header()?.ok_or(JwCryptoError::IllegalState(
         "protected_header must be present.",
     ))?;
```

Comment:
This `protected_header` check is now done by `aes::aes_gcm_decrypt` so you don't need to repeat it here. (Although if you wanted to check that there was no `jwe.encrypted_key()` field, this would be the other place to do it).